src/models/sourcecodemodel.cpp

```diff
@@ -230,3 +230,46 @@ void SourceCodeModel::setSysroot(const QString& sysroot)
 {
     m_sysroot = sysroot;
 }
+
+void SourceCodeModel::findForward(const QString& search, const QModelIndex& start, bool caseSensitive)
+{
+    const auto searchString = caseSensitive ? search : search.toLower();
+    for (int i = 1; i < m_numLines; i++) {
+        // start searching at start and wrap around when you reach the end
+        int lineIndex = i + (start.isValid() ? start.row() : 0);
+        lineIndex %= m_numLines;
+
+        if (searchLine(lineIndex, searchString, caseSensitive)) {
+            emit searchResultFound(index(lineIndex, SourceCodeColumn));
+            return;
+        }
+    }
+    emit noSearchResult();
+}
+
+void SourceCodeModel::findBackwards(const QString& search, const QModelIndex& start, bool caseSensitive)
```

Comment:
this is basically the same code as above, the only difference is the direction to go from the current line. please extract common code out and share code